connect/src/routes/route.ts

```diff
@@ -14,19 +16,54 @@ export type ValidationResult<OP> =
   | { params: ValidatedTransferParams<OP>; valid: true }
   | { params: TransferParams<OP>; valid: false; error: Error };
 
-export type RouteConstructor<N extends Network, OP> = {
+// Custom options to use for the transfer
+export interface Options {
+  // TODO
+}
+
+export interface Receipt {
+  from: Chain;
+  to: Chain;
+  originTxs?: TransactionId[];
+  state: TransferState;
+}
+
+export interface Quote {
+  sourceToken: {
+    token: TokenId;
+    amount: bigint;
+  };
+  destinationToken: {
+    token: TokenId;
+    amount: bigint;
```

Comment:
do we need to complicate this any further for swaps? swaps usually will give a minimum amount (worst case), and an estimated amount. I suppose they could just add a `minimumAmount` field to their quote, right? should we add a `minimumAmount?: bigint` here? wdyt